openvpn/primitives/ca.go

```diff
@@ -0,0 +1,160 @@
+package primitives
+
+import (
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"fmt"
+	log "github.com/cihub/seelog"
+	"math/big"
+	"os"
+	"time"
+)
+
+// CreateCA creates Certificate Authority certificate and private key
+func (p *SecurityPrimitives) CreateCA() (*x509.Certificate, error) {
+	log.Info("Create CA (", p.CACertPath, ", ", p.CAKeyPath, ")")
+
+	ca := &x509.Certificate{
+		SerialNumber: big.NewInt(1653),
+		Subject: pkix.Name{
+			Country:            []string{"Gibraltar"},
```

Comment:
These needs to be moved at least to const level first, maybe extracting this info as parameters